utils/pod_cidr.go

```diff
@@ -41,17 +59,53 @@ func InsertPodCidrInCniSpec(cniConfFilePath string, cidr string) error {
 	if err != nil {
 		return fmt.Errorf("Failed to load CNI conf file: %s", err.Error())
 	}
-	config := make(map[string]interface{})
-	err = json.Unmarshal(file, &config)
-	if err != nil {
-		return fmt.Errorf("Failed to parse JSON from CNI conf file: %s", err.Error())
-	}
+	if strings.HasSuffix(cniConfFilePath, ".conflist") {
+		var config interface{}
+		err = json.Unmarshal(file, &config)
+		if err != nil {
+			return fmt.Errorf("Failed to parse JSON from CNI conf file: %s", err.Error())
+		}
+		updatedCidr := false
+		configMap := config.(map[string]interface{})
+		for key := range configMap {
+			if key != "plugins" {
+				continue
+			}
+			// .conflist file has array of plug-in config. Find the one with ipam key
+			// and insert the CIDR for the node
+			pluginConfigs := configMap["plugins"].([]interface{})
+			for _, pluginConfig := range pluginConfigs {
+				pluginConfigMap := pluginConfig.(map[string]interface{})
+				if val, ok := pluginConfigMap["ipam"]; ok {
+					valObj := val.(map[string]interface{})
+					valObj["subnet"] = cidr
+					updatedCidr = true
+					break
+				}
+			}
+		}
 
-	config["ipam"].(map[string]interface{})["subnet"] = cidr
-	configJson, _ := json.Marshal(config)
-	err = ioutil.WriteFile(cniConfFilePath, configJson, 0644)
-	if err != nil {
-		return fmt.Errorf("Failed to insert subnet cidr into CNI conf file: %s", err.Error())
+		if !updatedCidr {
+			return fmt.Errorf("Failed to insert subnet cidr into CNI conf file: %s as CNI file is invalid.", cniConfFilePath)
+		}
+
+		configJSON, _ := json.Marshal(config)
+		err = ioutil.WriteFile(cniConfFilePath, configJSON, 0644)
+		if err != nil {
+			return fmt.Errorf("Failed to insert subnet cidr into CNI conf file: %s", err.Error())
+		}
+	} else {
+		config := make(map[string]interface{})
+		err = json.Unmarshal(file, &config)
+		if err != nil {
+			return fmt.Errorf("Failed to parse JSON from CNI conf file: %s", err.Error())
+		}
+		config["ipam"].(map[string]interface{})["subnet"] = cidr
+		configJSON, _ := json.Marshal(config)
```

Comment:
Seems like these three lines can be added after the if/else 
```go
configJSON, _ := json.Marshal(config)
err = ioutil.WriteFile(cniConfFilePath, configJSON, 0644)
if err != nil {
     return fmt.Errorf("Failed to insert subnet cidr into CNI conf file: %s", err.Error())
}
```

Reply:
Done!